`.conda` packages in `repodata.json`

olenagerasimova opened the pull request:
Part of #5 
It appears that .tar.bz2 and .conda packages are separated, the `repodata.json` format is:
```
{ 
  packages: { \\all tar.bz2 packages listed here\\ },
  packages.conda: { \\all .conda packages listed here\\ }
}
```
I've updated readme, corrected classes and tests.

g4s8 commented on src/test/com/artipie/conda/CondaRepodataRemoveTest.java:
```diff
@@ -36,22 +39,21 @@ void removesPackagesInfo() throws IOException {
+                        "\"md5\":\"0ebe0cb0d62eae6cd237444ba8fded66\",",
+                        "\"name\":\"decorator\",",
                         // @checkstyle LineLengthCheck (1 line)
-                        "\"sha256\":\"1fe3c3f4250e51886838e8e0287e39076d601b9f493ea05c37a2630a9fe5810f\",",
-                        "\"size\":123,",
-                        "\"subdir\":\"macOS\",",
-                        "\"timestamp\":153073163434,",
-                        "\"version\":\"0.5.0\"",
-                        "}",
-                        "}",
-                        "}"
+                        "\"sha256\":\"b5f77880181b37fb2e180766869da6242648aaec5bdd6de89296d9dacd764c14\",",
+                        "\"size\":15638,",
+                        "\"subdir\":\"linux-64\",",
+                        "\"timestamp\":1516376429792,",
+                        "\"version\":\"4.2.1\"",
+                        "}}",
+                        ",\"packages.conda\":{}}"
```
Checking JSON against string representation is not really robust: e.g. this test can break in case of library updates or CI system environment changes, when JSON library produces the same valid JSON object but change encoding or formatting. I'd recommend using json matchers like in other adapters

g4s8 commented on src/test/com/artipie/conda/meta/JsonMaidTest.java:
```diff
@@ -18,25 +18,35 @@
                     "\n", "{", "  \"packages\" : {",
-                    String.format("%s,\n%s", this.funPackageZeroOne(), this.funPackageZeroTwo()),
-                    "  }", "}"
+                    String.format("%s,\n%s", this.cram(), this.decorator()),
+                    "  },",
+                    "  \"packages.conda\" : {",
+                    this.tenacity(),
+                    "  }",
+                    "}"
```
And same here. Aside from it, this assertion is hard to read and understand.

g4s8 commented on src/test/com/artipie/conda/meta/JsonMaidTest.java:
```diff
@@ -71,16 +84,72 @@ void removesFirstPackage() throws IOException {
-                    "  \"packages\" : {",
-                    String.format("%s,\n%s", this.testPackage(), this.funPackageZeroTwo()),
-                    "  }",
+                    "\n", "{", "  \"packages\" : {",
+                    String.format("%s,\n%s", this.decorator(), this.pyqt()),
+                    "  },",
+                    "  \"packages.conda\" : {",
+                    String.format("%s,\n%s", this.notebook(), this.tenacity()),
+                    "  }", "}"
```
And same here

g4s8 commented on src/test/com/artipie/conda/meta/JsonMaidTest.java:
```diff
@@ -71,16 +84,72 @@ void removesFirstPackage() throws IOException {
+        MatcherAssert.assertThat(
+            stream.toString(),
+            new IsEqual<>(
+                String.join(
+                    "\n", "{", "  \"packages\" : {",
+                    String.format("%s,\n%s,\n%s", this.cram(), this.decorator(), this.pyqt()),
+                    "  },",
+                    "  \"packages.conda\" : { }",
```
And here

g4s8 commented on src/test/com/artipie/conda/meta/JsonMaidTest.java:
```diff
@@ -117,61 +186,118 @@ void doesNothingIfChecksumsAreEmpty() throws IOException {
     }
 
-    private String funPackageZeroTwo() {
+    private String cram() {
```
This method could be converted to `private static final String` field - it's always the same

g4s8 commented on src/test/com/artipie/conda/meta/JsonMaidTest.java:
```diff
@@ -117,61 +186,118 @@ void doesNothingIfChecksumsAreEmpty() throws IOException {
+        );
+    }
+
+    private String decorator() {
```
This method is also could be refactored to field

g4s8 commented on src/test/com/artipie/conda/meta/JsonMaidTest.java:
```diff
@@ -117,61 +186,118 @@ void doesNothingIfChecksumsAreEmpty() throws IOException {
     }
 
-    private String funPackageZeroOne() {
+    private String pyqt() {
```
And this method

g4s8 commented on src/test/com/artipie/conda/meta/JsonMaidTest.java:
```diff
@@ -117,61 +186,118 @@ void doesNothingIfChecksumsAreEmpty() throws IOException {
+        );
+    }
+
+    private String notebook() {
```
And this method. Maybe it'd be easier to maintain some test resources with JSON files instead of putting JSON data into test code, WDYT?

g4s8 commented on src/test/com/artipie/conda/meta/JsonMaidTest.java:
```diff
@@ -117,61 +186,118 @@ void doesNothingIfChecksumsAreEmpty() throws IOException {
     }
 
-    private String testPackage() {
+    private String tenacity() {
```
Same here

g4s8 reviewed (changes requested):
@olenagerasimova please check my comments

olenagerasimova commented:
@g4s8 thanks, I've added dependency to compare jsons (in our case - ignoring formatting and order) and extracted files for `JsonMaidTest` assertions. As for `CondaRepodata` I think it's ok to leave resulting json as is in the assertion.

olenagerasimova commented:
@g4s8 please, take a look, I need it merged to continue the work

g4s8 reviewed (approved):
@olenagerasimova thanks
